### PR TITLE
Issue/124: Upgrade typer to 0.4.0 to support click 8.0.0

### LINKED ISSUE #124: cruft 2.9.0 on Pypi requires click <= 8.0.0
The setup.py shipped on Pypi has an old version of click fixed as depdency. I cannot find any reference in the repo for that requirement. can you please update the tarball?

### PR DESCRIPTION
**Pull request by dmitchell7**
typer 0.4.0 adds support for click 8 and maintains compatibility with click 7.

fixes #124

### CODE REVIEW

**Comment by dmitchell7** [comment]
`poetry update`, `poetry install` and `pytest` all succeed.

```
$ poetry update
Updating dependencies
Resolving dependencies... (105.4s)

Writing lock file

Package operations: 94 installs, 0 updates, 0 removals

  • Installing six (1.16.0)
  • Installing certifi (2021.5.30)
  • Installing charset-normalizer (2.0.4)
  • Installing idna (3.2)
  • Installing markupsafe (2.0.1)
  • Installing pyparsing (2.4.7)
  • Installing python-dateutil (2.8.2)
  • Installing pyyaml (5.4.1)
  • Installing urllib3 (1.26.6)
  • Installing zipp (3.5.0)
  • Installing attrs (21.2.0)
  • Installing click (8.0.1)
  • Installing falcon (2.0.0)
  • Installing ghp-import (2.0.1)
  • Installing importlib-metadata (4.8.1)
  • Installing jinja2 (3.0.1)
  • Installing markdown (3.3.4)
  • Installing mccabe (0.6.1)
  • Installing mergedeep (1.3.4)
  • Installing more-itertools (8.9.0)
  • Installing packaging (21.0)
  • Installing pluggy (0.13.1)
  • Installing py (1.10.0)
  • Installing pycodestyle (2.7.0)
  • Installing pyflakes (2.3.1)
  • Installing pyyaml-env-tag (0.1)
  • Installing requests (2.26.0)
  • Installing smmap (4.0.0)
  • Installing watchdog (2.1.5)
  • Installing wcwidth (0.2.5)
  • Installing arrow (1.1.1)
  • Installing chardet (4.0.0)
  • Installing decorator (5.0.9)
  • Installing docstring-parser (0.7.3)
  • Installing flake8 (3.9.2)
  • Installing gitdb (4.0.7)
  • Installing hug (2.6.1)
  • Installing ipython-genutils (0.2.0)
  • Installing mako (1.1.5)
  • Installing mkdocs (1.2.2)
  • Installing mkdocs-material-extensions (1.0.3)
  • Installing parso (0.8.2)
  • Installing pbr (5.6.0)
  • Installing ptyprocess (0.7.0)
  • Installing pygments (2.10.0)
  • Installing pymdown-extensions (7.1)
  • Installing pytest (5.4.3)
  • Installing text-unidecode (1.3)
  • Installing toml (0.10.2)
  • Installing tornado (6.1)
  • Installing typing-extensions (3.10.0.2)
  • Installing appdirs (1.4.4)
  • Installing backcall (0.2.0)
  • Installing binaryornot (0.4.4)
  • Installing coverage (5.5)
  • Installing dparse (0.5.1)
  • Installing execnet (1.9.0)
  • Installing flake8-polyfill (1.0.2)
  • Installing gitpython (3.1.20)
  • Installing jedi (0.18.0)
  • Installing jinja2-time (0.2.0)
  • Installing livereload (2.6.3)
  • Installing mkdocs-material (7.2.6)
  • Installing mypy-extensions (0.4.3)
  • Installing pathspec (0.9.0)
  • Installing pdocs (1.1.1)
  • Installing pexpect (4.8.0)
  • Installing pickleshare (0.7.5)
  • Installing poyo (0.5.0)
  • Installing prompt-toolkit (3.0.3)
  • Installing pydantic (1.7.4)
  • Installing pytest-forked (1.3.0)
  • Installing python-slugify (5.0.2)
  • Installing regex (2021.8.28)
  • Installing stevedore (3.4.0)
  • Installing traitlets (4.3.3)
  • Installing typed-ast (1.4.3)
  • Installing yaspin (0.15.0)
  • Installing bandit (1.7.0)
  • Installing black (20.8b1)
  • Installing cookiecutter (1.7.3)
  • Installing examples (1.0.2)
  • Installing flake8-bugbear (19.8.0)
  • Installing ipython (7.16.1)
  • Installing isort (5.8.0)
  • Installing mypy (0.812)
  • Installing pep8-naming (0.8.2)
  • Installing portray (1.7.0)
  • Installing pytest-cov (2.12.1)
  • Installing pytest-mock (3.6.1)
  • Installing pytest-xdist (1.34.0)
  • Installing safety (1.10.3)
  • Installing typer (0.4.0)
  • Installing vulture (1.6)
$ poetry install
Installing dependencies from lock file

No dependencies to install or update

Installing the current project: cruft (2.9.0)
$ pytest
========================================================================================================================================= test session starts ==========================================================================================================================================
platform linux -- Python 3.9.2, pytest-5.4.3, py-1.10.0, pluggy-0.13.1
rootdir: /home/dmitchell, inifile: setup.cfg
plugins: xdist-1.34.0, forked-1.3.0, cov-2.12.1, mock-3.6.1
collected 67 items                                                                                                                                                                                                                                                                                     

tests/test_api.py ........................                                                                                                                                                                                                                                                       [ 35%]
tests/test_cli.py ..............................                                                                                                                                                                                                                                                 [ 80%]
tests/test_exceptions.py .....                                                                                                                                                                                                                                                                   [ 88%]
tests/test_utils.py ........                                                                                                                                                                                                                                                                     [100%]

==================================================================================================================================== 67 passed in 103.07s (0:01:43) ====================================================================================================================================
```

File: pyproject.toml [hunk 1]
```diff
@@ -4,15 +4,15 @@ version = "2.9.0"
 description = "Allows you to maintain all the necessary cruft for packaging and building projects separate from the code you intentionally write. Built on-top of CookieCutter."
 authors = ["Timothy Crosley <timothy.crosley@gmail.com>", "Sambhav Kothari <sambhavs.email@gmail.com>"]
 license = "MIT"
-readme = "README.md"
+	readme = "README.md"
```
Comment (samj1912) [inline pyproject.toml:7]:
```suggestion
readme = "README.md"
```

**Review by samj1912 — APPROVED** [review]
LGTM apart from that

**Comment by dmitchell7** [comment]
@samj1912 I apologize, I don't use poetry often and just ran `poetry update` in the original commit. That updated more packages than I needed to. I reran with just `poetry update typer click` in 6ee7c9a in order to only update the typer and click packages. The failed tests should hopefully pass now:

```
/home/dmitchell/cruft$ python -V
Python 3.7.9
/home/dmitchell/cruft$ poetry run mypy --ignore-missing-imports cruft/
Success: no issues found in 17 source files

```

**Comment by samj1912** [comment]
Thank you so much for this contribution! Very much appreciate it!

**Comment by dmitchell7** [comment]
> Thank you so much for this contribution! Very much appreciate it!

My pleasure, @samj1912 . What is cruft's policy for creating new releases on pypi?

**Pull request merged**